packages/app/universal-ts-utils/src/array/uniq.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { describe, expect, it } from 'vitest'
+import { uniq } from './uniq'
+
+describe('uniq', () => {
+  it('returns a new array of mixed primitive value without duplicates', () => {
+    const objectA = {}
+    const values = [1, 'a', Number.NaN, true, false, objectA, null, undefined]
+    const duplicateValues = [...values, ...values]
```

Comment:
note that this way of creating duplicates ensure that they are equal not just by value but also by reference.

in order to make behaviour more explicit, it may be helpful to have another test, where duplicates have same value but different reference. I think that will cause some of deduplication fail, which is fine, but something that users of the method should be aware of.

Reply:
Is this method intended to be user with objects? as Igor mentioned objects are compared by reference and wondering if this can be confusing.

If we don't want to use this method with objects maybe we can restrict the accepted types to avoid misunderstanding

Reply:
I added comment to clarify that the uniqueness is compared based on reference check.